src/transformers/modeling_tf_utils.py

```diff
@@ -301,6 +301,31 @@ def load_tf_weights(model, resolved_archive_file):
     K.batch_set_value(weight_value_tuples)
 
 
+def save_tf_model(model, save_directory):
+    """
+    Save a model to a directory.
+
+    Arguments:
+        model (:obj:`tf.keras.models.Model`):
+            Model to save.
+        save_directory (:obj:`str`):
+            Directory to which to save. Will be created if it doesn't exist.
+    """
+    if os.path.isfile(save_directory):
+        logger.error("Provided path ({}) should be a directory, not a file".format(save_directory))
+        return
+    os.makedirs(save_directory, exist_ok=True)
+
+    # If we save using the predefined names, we can load using `from_pretrained` if ``model`` is an instance of class `TFPreTrainedModel`.
+    output_model_file = os.path.join(save_directory, TF2_WEIGHTS_NAME)
+    model.save_weights(output_model_file)
+    logger.info("Model weights saved in {}".format(output_model_file))
+
+    # Also save to SavedModel format.
+    tf.saved_model.save(model, export_dir=save_directory)
```

Comment:
No, saving a `saved_model` must be done **only** in the trainer, and should not happen when calling `.save_pretrained()`.

Reply:
Hi, 

The modified `TFPreTrainedModel.save_pretrained` is actually equivalent to its previous form - if we don't add `tf.saved_model.save` part. The test failed only after I add this new part today.

Maybe I missed some requirement I don't know. I will try further to see if I can get something working by avoiding using tf.saved_model.save() inside a call to `save_pretrained()`.

Your code snippet is good, but I just tried to avoid the duplicated code like

```
    if os.path.isfile(save_directory):
        logger.error("Provided path ({}) should be a directory, not a file".format(save_directory))
        return
    os.makedirs(save_directory, exist_ok=True)
```

Reply:
The tests fails because now the saving part is outside. Make it back inside the class and the tests will pass. Having few duplicated lines of code is not a big deal.

Reply:
It turns out that it works after I removed

```
        # Also save to SavedModel format.
        tf.saved_model.save(self.model, export_dir=output_dir)
        logger.info("SavedModel saved in {}".format(output_dir))
```

to `TFTrainer.save_model()`. Would this current version acceptable, @jplu ?

Reply:
Yep, this is normal, it is because the saved model TF function could not find the proper serialization for the parameters.

Can you just put back as it was, please?

Reply:
OK, I could do that. But would you mind to give me some more explanation? The current version moves `self.save_weights` outside `TFPreTrainedModel`, but it should be the same thing. And current version passes the tests, so there shouldn't be any problem.

Sorry to bother, but it is good to know what's going on here. Thanks for your patience.

Reply:
No worries it is ok :)

It is mostly because I'm currently reworking a lot of things of TF side including the way we save/load the weights. So the less changes on this part the less updates I will do later 😄

Reply:
OK, I got it now :) I will rollback to its previous form, no problem. Thanks again.